x-pack/platform/plugins/shared/fleet/server/services/agents/change_privilege_level.ts

```
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, SavedObjectsClientContract } from '@kbn/core/server';
+
+import { FleetUnauthorizedError } from '../../errors';
+import { packagePolicyService } from '../package_policy';
+
+import { createAgentAction } from './actions';
+import { getAgentById } from './crud';
+
+export async function changeAgentPrivilegeLevel(
+  esClient: ElasticsearchClient,
+  soClient: SavedObjectsClientContract,
+  agentId: string,
+  options: {
+    unprivileged: boolean;
```

should we return a 400 error if `unprivileged: false` is used, as it is not supported? https://github.com/elastic/kibana/issues/221889#issuecomment-3209732427

---

Yes, that should be clearer 👍 Maybe with an error message that says that granting privilege is not currently supported. Alternatively, we could remove `unprivileged` from user-facing options altogether (and pass it with value `true` in the action body) and only leave `user_info`, WDYT?